File: old/actions/install_dotnet.py
"""
Install .NET SDK action
"""
import logging
from cli import Apt
from .base import Action
logger = logging.getLogger(__name__)

class InstallDotnetAction(Action):
    """Action to install .NET SDK"""
    description = "dotnet installation"

    def execute(self) -> bool:
        """Install .NET SDK"""
        if not self.apt_service:
            logger.error("APT service not initialized")
            return False
        logger.info("Installing .NET SDK...")
        install_cmd = Apt().install(["dotnet-sdk-8.0"])
        output = self.apt_service.execute(install_cmd)
        if output is None:
            logger.error("dotnet SDK installation failed")
            return False
        return True

File: src/actions/install_dotnet.go
package actions

import (
	"enva/libs"
	"enva/services"
)

// InstallDotnetAction installs .NET SDK
type InstallDotnetAction struct {
	*BaseAction
}

func NewInstallDotnetAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallDotnetAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallDotnetAction) Description() string {
	return "dotnet installation"
}

func (a *InstallDotnetAction) Execute() bool {
	if a.APTService == nil {
		libs.GetLogger("install_dotnet").Printf("APT service not initialized")
		return false
	}
	libs.GetLogger("install_dotnet").Printf("Installing .NET SDK...")
	output, exitCode := a.APTService.Install([]string{"dotnet-sdk-8.0"})
	if exitCode == nil || *exitCode != 0 {
		libs.GetLogger("install_dotnet").Printf("dotnet SDK installation failed: %s", output)
		return false
	}
	return true
}

